### PR TITLE
 Upgrade CLI to v2 and added arbitrary ranges

### PR DESCRIPTION
**Pull request by mfranzil**
See issue #15.

This PR improves on the tool by:
- updating to Go 1.18,
- updating the urfave/cli package to v2 (with relative changes)
- adding the new Go files to .gitignore,

and most importantly, introduces the possibility of providing a start interval and a duration, rather than a duration which is calculated backwards from the current time. Might need some extra work (as I used a questionable approach for the `steps` function in `prometheus.go`), and probably the vendoring could also be removed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ Export Prometheus data as CSV or directly plot with gnuplot & matplotlib.
 ## Installation
 
 ```
-go get -v -u github.com/go-pluto/styx
+go install github.com/mfranzil/styx@latest
```
Comment (metalmatze) [inline README.md]: That needs to be undone.

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,20 @@
+module github.com/mfranzil/styx
```
Comment (metalmatze) [inline go.mod]: That needs to be updated to the correct path.

File: go.mod [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+go 1.18
```
Comment (metalmatze) [inline go.mod]: We can already go for 1.20, I guess.

File: prometheus.go [hunk 4]
```diff
@@ -88,13 +88,7 @@ func Query(host string, start time.Time, end time.Time, query string) ([]Result,
 }
 
 func steps(dur time.Duration) int {
```
Comment (metalmatze) [inline prometheus.go:90]: Hm, this might result in very high amount data, for exporting data with 1w (or even longer) time ranges.

**Review by metalmatze — COMMENTED** [review]
Thank you for posting it!
There are a couple of things we should address before merging the PR. Overall it looks really good!

**Comment by mfranzil** [comment]
I have fixed the first three issues (and in the meantime, also removed a legacy feature of the old cli v1 API, we can just obtain the parameters without creating structs)

However, I'm kind of unsure how to fix the last one. Should we add options for granularity? We should also probably be more transparent to the user when diminishing it. When I first used the tool, I was confused why having long time ranges had such a low amount of data points.

**Comment by metalmatze** [comment]
Yeah, it all depends on the expectation. What we usually do in UI is to have the duration be a pixel on the canvas. For example, given you have 1000 pixel width, for one week we'd do `(7*24*60*60)/1000 ~ 600s = 10m`. Now, this doesn't really work here. We could get the raw output, or have a parameter that specifies how many datapoints someone expects. 

Either way, it seems we can do that one on a follow-up PR.

Thank you so much for these changes!

**Review by metalmatze — APPROVED** [review]

**Comment by mfranzil** [comment]
> Yeah, it all depends on the expectation. What we usually do in UI is to have the duration be a pixel on the canvas. For example, given you have 1000 pixel width, for one week we'd do `(7*24*60*60)/1000 ~ 600s = 10m`. Now, this doesn't really work here. We could get the raw output, or have a parameter that specifies how many datapoints someone expects.
> 
> Either way, it seems we can do that one on a follow-up PR.
> 
> Thank you so much for these changes!

See #18 as a follow-up. You're welcome!
(Also sorry if I respond so late, I tend to procrastinate stuff in my inbox too much ahah)

**Pull request merged**